overlays/tools.nix

```diff
@@ -0,0 +1,152 @@
+# A loosely curated set of tools we can select with:
+#   haskell-nix.tools.cabal
+# or
+#   shellFor { tools = ts: with ts; [ cabal ]; }
+#
+# It will be compiled with the same version of ghc used in the
+# shell (the build ghc in the case of cross compilation).
+#
+# If want to specify the ghc used to compile. Use:
+#   haskell-nix.tools.ghc883.cabal
+# or
+#   shellFor { tools = ts: with ts; [ ghc883.cabal ]; }
+#
+# By default we will get the latest version of the tool.
+# To specify the version use:
+#   haskell-nix.tools.cabal."3.2.0.0"
+# or
+#   shellFor { tools = ts: with ts; [ cabal."3.2.0.0" ]; }
+#
+# To add a new tool to this file:
+#   * Add it to `hackageToolVersions`.
+#   * Test it with:
+#       nix-build -E 'let h = (import ./. {}); in (import h.sources.nixpkgs-default h.nixpkgsArgs).haskell-nix.tools.ghc865.cabal."3.2.0.0"'
+#   * If necessary add `modules` it needs to `hackageToolModules`.
+#
+# To add a new tool that is not in hackage:
+#   * Add it to `otherTools` in `toolsForGhc`.
+#
+# To include materialization for a tool:
+#   * Build the tool with `checked-tools` instead of `tools`
+#     (it will fail to build since the materialized
+#     files are missing):
+#       nix-build -E 'let h = (import ./. {}); in (import h.sources.nixpkgs-default h.nixpkgsArgs).haskell-nix.checked-tools.ghc865.cabal."3.2.0.0"'
+#   * Look in the logs for something like:
+#       Materialized nix used for cabal-install-plan-to-nix-pkgs is missing. To fix run :
+#         cp -r /nix/store/dwhg7djxwzd0bg69m3fqhywk7fbyf01i-cabal-install-plan-to-nix-pkgs /Users/hamish/iohk/haskell.nix/materialized/tools/ghc-8.6.5-cabal-3.2.0.0
+#         chmod -R +w /Users/hamish/iohk/haskell.nix/materialized/tools/ghc-8.6.5-cabal-3.2.0.0
+#   * Run the two commands from the log
+#   * Run `git add materialized` before committing the changes.
+self: super:
+let
+  inherit (self) lib;
+
+  # Tools and versions to include from hackage
+  hackageToolVersions = {
+    # Default is version the head of the list
+    cabal           = ["3.2.0.0" "3.0.0.0" "2.4.1.0"];
```

Comment:
Does the cabal version we build work? In particular, `cabal test`: https://github.com/input-output-hk/haskell.nix/issues/422

Reply:
What should I run in the plutus nix-shell to check?

Reply:
I think `cabal test language-plutus-core` should do.